R/01-scipiper.R

```diff
@@ -11,9 +11,78 @@
   op <- options()
   op.scipiper <- list(
     scipiper.remake_file = 'remake.yml',
-    scipiper.gd_config_file = "lib/cfg/gd_config.yml"
+    scipiper.gd_config_file = "lib/cfg/gd_config.yml",
+    scipiper.s3_config_file = "lib/cfg/s3_config.yml",
+    scipiper.ind_ext = 'ind'
   )
   toset <- !(names(op.scipiper) %in% names(op))
   if(any(toset)) options(op.scipiper[toset])
   
 }
+
+#' Scipiper session options
+#'
+#' Several session-level options are stored in R's `.Options` object.
+#'
+#' \describe{
+#'
+#' \item{`scipiper.remake_file`}{File path of the main remake YAML file for a
+#' project. Used as the default for the `remake_file` argument in many
+#' functions.}
+#'
+#' \item{`scipiper.s3_config_file`}{File path to a configuration file for an
+#' Amazon S3 bucket. Used as the default for the `s3_config_file` argument in
+#' many functions (usually those prefixed with `s3_`).}
+#'
+#' \item{`scipiper.gd_config_file`}{File path to a configuration file for a
+#' Google Drive folder. Used as the default for the `gd_config_file` argument in
+#' many functions (usually those prefixed with `gd_`).}
+#'
+#' \item{`scipiper.ind_ext`}{The indicator file extension to recognize, i.e.,
+#' the final file extension of files for which `is_ind_file()` should return
+#' `TRUE` and which `as_ind_file` should append to the data file name. Used as
+#' the default for the `ind_ext` argument in many functions.`}
+#'
+#' }
+#'
+#' @docType package
+#' @name options
+#' @md
```

Comment:
what's this?

Reply:
https://cran.r-project.org/web/packages/roxygen2/vignettes/markdown.html

Reply:
Huh! Never noticed that option. Are you just putting that in by habit, or is there some markdown in there that I'm not noticing? If by habit...the link you just sent me to says you can put this:
```
Roxygen: list(markdown = TRUE)
```
in the DESCRIPTION

Seems like if you are using markdown...you could simplify that item list to be markdowny (not saying you need to....just getting excited about using markdown in my future docs.....)

Reply:
Nice, thanks for pointing that out. I started using `@md` when introducing markdown to more developed packages, but scipiper seems like a good place to use the package-wide option. Switching over right now...

Reply:
Switched the `@md` tags. Just noticed that I could probably switch the itemized list, too, but i'm gonna pass on that for now because I should be finishing other tasks today. Another time